### PR TITLE
feat: add Select component

### PR DESCRIPTION
**Pull request by korvin89**
Add `Select` component (#16)

### CODE REVIEW

File: src/components/unstable/Select/components/Control/Control.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        >
+            {label && <span className={b('label')}>{label}</span>}
+            {showPlaceholder && <span className={b('placeholder')}>{placeholder}</span>}
+            {showOptionsText && <span className={b('option-text')}>{optionsText.join(', ')}</span>}
+            <Icon className={b('chevron-icon')} data={Chevron} />
+        </button>
+    );
+});
+
+Control.displayName = 'Control';
```
Comment (amje) [inline src/components/unstable/Select/components/Control/Control.tsx:129]: SelectControl?

File: src/components/unstable/Select/tech-components.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import {SelectOption, SelectOptgroup} from './types';
+
+export const Option: React.FC<SelectOption> = () => null;
+
+export const Optgroup: React.FC<SelectOptgroup> = () => null;
```
Comment (amje) [inline src/components/unstable/Select/tech-components.tsx:5]: Lets name this as `OptionGroup`

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/unstable/Select/__stories__/SelectShowcase.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    EXAMPLE_CHILDREN_OPTIONS,
+    EXAMPLE_GROUP_JSON_OPTIONS,
+    EXAMPLE_GROUP_CHILDREN_OPTIONS,
+    EXAMPLE_CUSTOM_OPTIONS,
+} from './constants';
+
+import './SelectShowcase.scss';
+
+const b = block('select-showcase');
+const MODE = {
```
[the remark this answers is not in the document]
Comment (korvin89) [reply src/components/unstable/Select/__stories__/SelectShowcase.tsx:18]: There is no need to use enum in this case. We don`t need type. Also enum transpiling into horrible js construction

File: src/components/unstable/Select/utils.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ export const getFlattenOptions = (options: (SelectOption | SelectOptgroup)[]): F
     }, [] as FlattenOption[]);
 };
 
-export const getListboxItemHeight = (args: {
+export const getSelectPopupItemHeight = (args: {
```
Comment (amje) [inline src/components/unstable/Select/utils.ts:35]: "Select" part can be omitted like in the function below `getPopupVerticalOffset `

File: src/components/unstable/Select/utils.ts [hunk 5]
```diff
@@ -46,19 +46,22 @@ export const getListboxItemHeight = (args: {
     return getOptionHeight ? getOptionHeight(option) : SIZE_TO_ITEM_HEIGHT[size];
 };
 
-export const getListboxHeight = (args: {
+export const getSelectPopupHeight = (args: {
```
Comment (amje) [inline src/components/unstable/Select/utils.ts:51]: same

File: src/components/unstable/Select/utils.ts [hunk 6]
```diff
@@ -46,19 +46,22 @@ export const getListboxItemHeight = (args: {
...
     return options.reduce((height, option, index) => {
-        return height + getListboxItemHeight({getOptionHeight, size, option, index});
+        return height + getSelectPopupItemHeight({getOptionHeight, size, option, index});
     }, 0);
 };
 
-export const getPopupVerticalOffset = (args: {listboxHeight: number; controlRect?: DOMRect}) => {
-    const {listboxHeight, controlRect} = args;
+export const getPopupVerticalOffset = (args: {
+    selectPopupHeight: number;
```
Comment (amje) [inline src/components/unstable/Select/utils.ts:63]: just popupHeight will be enough

File: src/components/unstable/Select/components/SelectControl/SelectControl.scss [hunk 7]
```diff
@@ -20,6 +20,11 @@ $block: '.#{$ns}select';
 #{$block} {
     @include button-reset();
     @include button-text-color(var(--yc-color-text-primary));
+    @include pin(
+        $block,
+        (#{$block}_size_s, #{$block}_size_m, #{$block}_size_l, #{$block}_size_xl),
+        var(--yc-button-border-radius)
```
Comment (amje) [inline src/components/unstable/Select/components/SelectControl/SelectControl.scss:26]: button -> select

File: src/components/unstable/Select/components/SelectControl/SelectControl.scss [hunk 8]
```diff
@@ -20,6 +20,11 @@ $block: '.#{$ns}select';
 #{$block} {
     @include button-reset();
     @include button-text-color(var(--yc-color-text-primary));
+    @include pin(
+        $block,
+        (#{$block}_size_s, #{$block}_size_m, #{$block}_size_l, #{$block}_size_xl),
```
Comment (amje) [inline src/components/unstable/Select/components/SelectControl/SelectControl.scss:25]: This results in four exact chunks of code. Just `$block` will be enough

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/unstable/Select/__stories__/SelectShowcase.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (amje) [reply src/components/unstable/Select/__stories__/SelectShowcase.tsx:18]: Then it's a namespace for constants and should be named as `Mode`. `MODE.CODE` looks a bit screamy.

File: src/components/unstable/Select/__stories__/SelectShowcase.scss [hunk 9]
```diff
@@ -0,0 +1,35 @@
+@import '../../../variables';
+
+$block: '.#{$ns}select-showcase';
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/unstable/Select/__stories__/SelectShowcase.scss:3]: ok, but it is not necessary for demos, `bem-cn-lite` could be used directly here

**Review by amje — APPROVED** [review]

**Pull request merged**